### PR TITLE
new version 2.0.0

### LINKED ISSUE #9: Change subscriber module callbacks by a function callback
Replace to have function instead of a module with a callback

### PR DESCRIPTION
**Pull request by peillis**
fixes #9 #10 #11

### CODE REVIEW

File: lib/events_manager/consumer.ex [hunk 1]
```diff
@@ -160,25 +141,31 @@ defmodule EventsManager.Consumer do
...
-    case apply(state.consumer_module, :consume_event, [payload]) do
-      :ok ->
-        Logger.debug("[EventsManager] Sending ack for message #{tag}")
-        :ok = get_module("Basic").ack(state.channel, tag)
-        {:ok, :ack}
-
-      {:error, reason} ->
+    errors =
+      state.consumer_functions
+      |> Enum.map(& &1.(payload))
```
Comment (JoeZ99) [inline lib/events_manager/consumer.ex:150]: I don't undestand this. In the example provided in the README.md you defined consumer functions as:
```elixir
 consumers: %{
    "test_topic_1" => [&Test.MyModule.my_function/1],
    "test_topic_2" => [&Test.MyModule.my_another_function/1, &Test.MyOtherModule.my_function/1],
  },
```
so, those functions are already "anonymized" . when you do `Enum.map(& &1.(payload))` I understand it really is `Enum.map(&(&1.(payload))` where `&1` is the function defined in config, and the other `&` is to make it anonymous, but -at least in the README example- it's already anonymous! (unless `&` is idempotent , that is `&(&(&1+1)) = &(&1+1)` and you're just making sure)

**Review by JoeZ99 — CHANGES_REQUESTED** [review]
just clarify me the doubt with the `&` operator...

File: lib/events_manager/consumer.ex (hunk 1 above)
Comment (peillis) [reply lib/events_manager/consumer.ex:150]: You got me there. I really don't know. All I know is that it works like this, but maybe the other way? I don't know.

File: lib/events_manager/application.ex [hunk 2]
```diff
@@ -6,11 +6,18 @@ defmodule EventsManager.Application do
   use Application
 
   def start(_type, _args) do
+    connection_uri = Application.get_env(:events_manager, :connection_uri)
+    consumers = Application.get_env(:events_manager, :consumers, [])
```
Comment (agutierrezrodriguez) [inline lib/events_manager/application.ex:10]: this won't fail, but is inconsistent because consumers config expect a map and not a list.

**Review by agutierrezrodriguez — APPROVED** [review]

File: lib/events_manager/consumer.ex (hunk 1 above)
Comment (JoeZ99) [reply lib/events_manager/consumer.ex:150]: I don't know how you got there, to put `& &1.(payload)` (without parenthesis) I've tried to find any example online and I couldn't . where did you get it from? anyway, they're not idempotent, at least not in the trials I've made, but if it works it works!!!

**Review by JoeZ99 — APPROVED** [review]

**Pull request merged**